toc/rfc/rfc-draft-ipv6-dual-stack-for-cf.md

```diff
@@ -0,0 +1,200 @@
+
+The following functionality is required in each CPI that supports dual stack networking:
+
+1. Enable dual stack mode if two manual networks refer to the same subnet
+This is [done](https://github.com/cloudfoundry/bosh-aws-cpi-release/pull/174) for AWS, but pending for other hyperscalers and infrastructures.
+2. Depending on the decided configuration in the cloud config and director implementation the cpi should:
+   * Assign an IPv6 prefix to a single VM where possible. The size of the prefix to be assigned must be configurable. Assignment must be done from a known pool, i.e. a larger CIDR range for the deployment that is then subdivided for each VM. This pool is also configurable. The CPI is responsible of providing information about available CIDR ranges, verifying that configuration is consistent and communicating with the infrastructure to apply the necessary assignments and mappings.
+   * Try to attach the IPv6 prefix passed in from the director
+
+#### BOSH Agent
+
+The BOSH Agent that is running on BOSH managed VMs is ultimately responsible for applying operating system level configuration. This includes network interface configuration.
+
+For supporting dual stack networking, the BOSH agent must support the following:
+
+1. Allow the agent to accept having multiple network settings for the same network
+
+   The BOSH Agent has a limitation that allows only a single BOSH network to be assigned to a Network Interface Card (NIC) in a virtual machine. When multiple BOSH networks are defined for the same hyperscaler subnet,
+   This limitation [has been resolved with bosh-agent#344 already.](https://github.com/cloudfoundry/bosh-agent/pull/344)
+
+2. With the BOSH CPI being able to delegate IPv6 prefixes to single VMs, the BOSH agent MUST be able to assign such CIDRs to VMs or at least not interfere with the setup (e.g. https://github.com/cloudfoundry/bosh-agent/pull/347 ), if the target infrastructure allows.
+
+#### BOSH Stemcells
+
+BOSH stemcells up to Ubuntu Jammy Jellyfish have IPv6 support disabled at kernel level. While it can be enabled, the resulting configuration change requires a VM reboot.
+
+The goal is to keep the already enabled IPv6 support from Ubuntu upstream enabled.
+This will be the case starting with Ubuntu Noble stemcell. Enabling IPv6 in older stemcell releases such as “Ubuntu Jammy” might be seen as backwards incompatible change.
+
+This topic is being discussed and tracked via the [issue os-conf-release#71](https://github.com/cloudfoundry/os-conf-release/issues/71) and must be followed up with appropriate fixes.
+
+### Diego
+
+Currently, the Silk CNI is used to create an overlay network, but also to assign IP addresses for the application containers. This assignment is done from a centrally managed pool of IP addresses.
```

Comment:
Was Silk CNI work specifically not included in this RFC?

Reply:
Please also note the extended section on Silk that was added by @plamen-bardarov